File: AssessmentApplication.Web/ngApps/SalesOrder/Models/Address.js
function Address(addressId, addressLine1, addressLine2, city, stateProvinceCode, postalCode) {
	this.addressId = addressId;
	this.addressLine1 = addressLine1;
	this.addressLine2 = addressLine2;
	this.city = city;
	this.stateProvinceCode = stateProvinceCode;
	this.postalCode = postalCode;
	this.fullAddressHtml = [addressLine1, addressLine2, city, stateProvinceCode, postalCode]
		.filter(function(val) {
			if (val) {
				return val;
			}
		})
		.join("<br />");
}

File: AssessmentApplication.DataContracts/Address.cs
using System;
using Newtonsoft.Json;

namespace AssessmentApplication.DataContracts
{
    [JsonObject(MemberSerialization.OptIn)]
    [Serializable]
    public class Address
    {
        #region Public Properties

        [JsonProperty]
        public int AddressId { get; set; }

        [JsonProperty]
        public string AddressLine1 { get; set; }

        [JsonProperty]
        public string AddressLine2 { get; set; }

        [JsonProperty]
        public string City { get; set; }

        [JsonProperty]
        public string PostalCode { get; set; }

        [JsonProperty]
        public string StateProvinceCode { get; set; }

        #endregion Public Properties
    }
}
